apps/api/src/domain/models/bibliographic-reference/common/commands/register-digital-representation-of-bibiliographic-citation/register-digital-representation-of-bibliographic-citation.command.integration.spec.ts

```diff
@@ -0,0 +1,130 @@
+
+const dummyFsa = buildTestCommandFsaMap().get(
+    commandType
+) as CommandFSA<RegisterDigitalRepresentationOfBibliographicCitation>;
+
+const validFsa = clonePlainObjectWithOverrides(dummyFsa, {
+    payload: {
+        aggregateCompositeIdentifier: existingBibliographicReference.getCompositeIdentifier(),
+        digitalRepresentationResourceCompositeIdentifier:
+            existingDigitalText.getCompositeIdentifier(),
+    },
+});
+
+const buildValidCommandFSA = () => validFsa;
+
+describe(commandType, () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let _idRepository: IIdRepository;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let app: INestApplication;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app, databaseProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: testDatabaseName,
+            }));
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+            idManager,
+        };
+
+        // idRepository = new ArangoIdRepository(databaseProvider);
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    afterAll(async () => {
+        await app.close();
+
+        databaseProvider.close();
+    });
+
+    describe(`when the command is valid`, () => {
+        it(`should succeed`, async () => {
+            await assertCommandSuccess(assertionHelperDependencies, {
+                systemUserId: dummySystemUserId,
+                seedInitialState: async () => {
+                    await testRepositoryProvider.addFullSnapshot(
+                        new DeluxeInMemoryStore({
+                            [ResourceType.bibliographicReference]: [existingBibliographicReference],
+                            [ResourceType.digitalText]: [existingDigitalText],
+                        }).fetchFullSnapshotInLegacyFormat()
+                    );
+                },
+                buildValidCommandFSA,
+                checkStateOnSuccess: async () => {
+                    const searchResult = await testRepositoryProvider
+                        .forResource(ResourceType.bibliographicReference)
+                        .fetchById(validFsa.payload.aggregateCompositeIdentifier.id);
+
+                    expect(searchResult).not.toBeInstanceOf(InternalError);
+
+                    const updatedInstance = searchResult as IBibliographicReference;
+
+                    const reference =
+                        updatedInstance.digitalRepresentationResoruceCompositeIdentifier;
```

Comment:
(sp) `digitalRepresentationResoruceCompositeIdentifier`

Reply:
Not now.